ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -2514,20 +2635,20 @@ private StoreFindToken revalidateStoreFindToken(StoreFindToken token,
             revalidatedToken = generateTokenBasedOnResetKey(token, indexSegments);
           }
           metrics.journalBasedTokenResetCount.inc();
-          logger.info("Revalidated token {} because it is invalid for the index segment map", token);
+          logger.info("Index {}: Revalidated token {} because it is invalid for the index segment map", dataDir, token);
         }
         break;
       case IndexBased:
         // An index based token, but the offset is not in the segments, might be caused by the compaction
         if (!indexSegments.containsKey(offset)) {
-          if (ignoreResetKey || token.getResetKey() == null) {
-            revalidatedToken = new StoreFindToken();
-          } else {
+          if (!ignoreResetKey && token.getResetKey() != null) {
             // reset key is present in the token
             revalidatedToken = generateTokenBasedOnResetKey(token, indexSegments);
```

Comment:
Are we confident to enable this generateTokenBasedOnResetKey for IndexBased token? Although I'm not sure why previously we didn't enable it.

Reply:
No idea, it was never enabled in the past. I checked the logic for reset key, they are solid. I don't see a reason not to enable it, except for that resetting back to the beginning of the log is easy to reason and making sure it's correct.